src/script/lua_api/l_util.cpp

```diff
@@ -423,6 +425,31 @@ int ModApiUtil::l_get_version(lua_State *L)
 	return 1;
 }
 
+int ModApiUtil::l_sha1(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	std::string data = luaL_checkstring(L, 1);
```

Comment:
`luaL_checklstring` should be used in case binary data is sent (unwanted NULL termination).

Reply:
Good catch, updated.